### PR TITLE
[windows] add implementation of getSystemFQDN() for Windows

### PR DESCRIPTION
**Pull request by derekwbrown**
companion doc PR https://github.com/DataDog/datadog-agent/pull/2250

### CODE REVIEW

**Review by arbll — DISMISSED** [review]
LGTM,

https://docs.microsoft.com/en-us/windows/desktop/api/winsock/ns-winsock-hostent

hostent.Name should return the FQDN.

**Review by olivielpeau — CHANGES_REQUESTED** [review]
In general, when making a change that could potentially change the canonical hostname reported by the Agent (such as this change), we have to be very careful of the impact on existing users.

In this specific case, I think we should use a different config flag for this. Explanation:

The `hostname_fqdn` flag, when enabled, is meant to reproduce the behavior of Agent v5 wrt hostname resolution, and we want to switch the default to `true` at some point in the future (see https://github.com/DataDog/datadog-agent/blob/master/docs/agent/hostname-resolution.md).

On Windows, Agent v5 never used the FQDN (as returned by the `GetHostByName` syscall) as the Agent hostname, so if we introduce this change as-is, and still recommend users to use `hostname_fqdn: true` when upgrading to v6, the hostname reported will change, which is undesirable.

Whereas, if we use a different config flag (disabled by default for now), we can ask windows users who want the new behavior to enable the flag, while users upgrading from Agent v5 will keep their hostnames unchanged.

File: releasenotes/notes/add-fqdn-support-windows-5a9b960c9df7c7aa.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  
+enhancements:
+  - |
+    On Windows, the `hostname_fqdn` flag will now be honored, and the 
+    host reported by DataDog will be the fully qualified hostname.
```
Comment (olivielpeau) [inline releasenotes/notes/add-fqdn-support-windows-5a9b960c9df7c7aa.yaml]: nit: correct case is `Datadog` :p

File: pkg/util/hostname.go [hunk 2]
```diff
@@ -209,10 +210,16 @@ func GetHostname() (string, error) {
...
+	if runtime.GOOS != "windows" {
+		// REMOVEME: This should be removed in 6.5
+		if err == nil && !config.Datadog.GetBool("hostname_fqdn") && fqdn != "" && hostName == h && h != fqdn {
+			log.Warnf("DEPRECATION NOTICE: The agent resolved your hostname as '%s'. However starting from version 6.5, it will be resolved as '%s' by default. To enable the behavior of 6.5+, please enable the `hostname_fqdn` flag in the configuration. For more information: https://dtdg.co/flag-hostname-fqdn", h, fqdn)
+		}
+	} else { // OS is Windows
+		if err == nil && !config.Datadog.GetBool("hostname_fqdn") && fqdn != "" && hostName == h && h != fqdn {
+			log.Warnf("The agent resolved your hostname as '%s', and will be reported this way maintain compatibility with version 5. To enable reporting as %s, please enable the `hostname_fqdn` flag in the configuration. For more information: https://dtdg.co/flag-hostname-fqdn", h, fqdn)
```
Comment (olivielpeau) [inline pkg/util/hostname.go]: missing word: `this way to maintain`

**Review by olivielpeau — DISMISSED** [review]
just nits, other than that LGTM

**Review by olivielpeau — DISMISSED** [review]
Addressed the nits, this should be ready for a merge

**Review by arbll — COMMENTED** [review]
Two small nits

**Review by arbll — APPROVED** [review]

**Pull request merged**